likes/views.py

```diff
@@ -1,3 +1,36 @@
-# from django.shortcuts import render
+from rest_framework import viewsets
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
 
-# Create your views here.
+from likes.models import Dislike, Like
+from posts.models import Post
+
+
+class LikeView(viewsets.ViewSet):
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request) -> Response:
+        user = request.user
+        post = request.POST["post_id"]
+        if Like.objects.filter(post=post).filter(user=user):
+            like = Like.objects.get(user=user, post=post)
```

Comment:
This is an interesting logic :+1::+1::+1:, but let's better have `POST` request for setting a like, `DELETE` request for removing it (having the same call for both setting and removing something can lead to errors when the client app accidentally sends two requests instead of one)

Reply:
Having `is_deleted` flag is great for soft deleting, for example, posts. Here let's just remove the `like` when it is removed

Reply:
Also, there is a case when a user sets both like and dislike. A possible solution is to have a common LikeDislike model with a `is_like` boolean flag which shows if it is a like or a dislike.

Reply:
Done